docs/diagram.md

```diff
@@ -0,0 +1,197 @@
+Call2 --load--> Load2[Target Config: jobs/configurations/jdk*ver.groovy] --> DSL
+Call2 --call--> Load3[Script: common/config_regeneration.groovy] --creatJob--> DSL
+
+DSL --create--> DS1[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant]
+DSL --create--> DS2[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant]
+DSL --create--> DS3[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant]
+DSL --create--> DS4[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant]
+```
+
+## High level diagram on Jenkins build/test jobs interaction
+
+```mermaid
+flowchart  LR  
+WeeklyTimer --run job--> PipelineW[build-scripts/weekly-openjdk*ver-pipeline]
+PipelineW --multiple trigger by variants as Release releaseType--> PipelineN[build-scripts/openjdk*ver-pipeline]
+NightlyTimer --rrun job with Nightly releaseType--> PipelineN
+
+PipelineN --run downstream--> DS1[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant>]
+PipelineN --run downstream--> DS2[build-scripts/jobs/jdk*ver/jdk*ver-*os-*rch-*variant>]
+PipelineN --run downstream --> DS3[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant>]
+PipelineN --run downstream --> DS12[build-scripts/jobs/jdk*ver/jdk*ver-*os-*arch-*variant>]
+```
+
+## Mainflow logic of creation pipeline: openjdk\*ver-pipeline
+
+```mermaid
+flowchart TD 
+Call[build/openjdk_pipeline.groovy] --load script--> 
+Load1[build/common/build_base_file.groovy]
+ 
+Call --load config--> Load2[build/jobs/configurations/jdk*ver_pipeline_config.groovy]
+
+Call --sharedlib --> 
+Load3[Git repo: openjdk-jenkins-helper]
+
+Load1 --call_function--> Build[doBuild]
+Load2 --input--> Build
+Load3 --input--> Build
+
+Build --create_downstream_jobs --> 
+Done[Job: jdk*ver/job/jdk*ver-*os-*arch-*variant]
+```
+
+## Mainflow logic of creation job: "build-scripts/job/utils/job/pipeline_jobs_generator_jdk*ver"
+
+```mermaid
+flowchart TD 
+build/regeneration/build_job_generator.groovy --load--> build/common/import_lib.groovy --load--> jobs/configurations/jdk*_pipeline_config.groovy --load--> jobs/configurations/jdk*.groovy --load--> common/config_regeneration.groovy --call_function--> regenerate --call_function--> makeJob --call--> createJob --> DSL[jobDSL: common/create_job_from_template.groovy] --set--> scm[Git clone: ci-jenkins-pipelines] --set--> properties[Permission/logRotator/...]--> parameters
+```
+
+## Mainflow logic of build job: jobs/jdk\*ver/jdk\*ver-\*os-\*arch-\*variant
+
+```mermaid
+flowchart TD
+starter[kick_off_build.groovy] --load--> import[build/common/import_lib.groovy] --load--> Load1[build/common/openjdk_build_pipeline.groovy] --call_function--> Builder[build] 
+Builder --sharedlib --> 
+Load3[Git repo: openjdk-jenkins-helper] --> docker{build in docker}
+docker --true: run--> dockerbuild[docker.build] --> sign
+docker --false:call_function--> CallbuildScript[buildScripts] --> sign{enableSigner} --true:call--> sing[sign] --> testStage{enableTests}
```

Comment:
```suggestion
docker --false:call_function--> CallbuildScript[buildScripts] --> sign{enableSigner} --true:call--> sign[sign] --> testStage{enableTests}
```

Reply:
Also, double-check ... as I thought we should be running the smoke tests always (and the enableTests piece was to enable the AQA set of tests).